src/core/thread/mle_router.cpp

```diff
@@ -2502,6 +2511,16 @@ void MleRouter::SetLeaderWeight(uint8_t aWeight)
     mLeaderWeight = aWeight;
 }
 
+uint32_t MleRouter::GetLeaderPartitionId(void) const
+{
+	return mLeaderPartitionId;
+}
+
+void MleRouter::SetLeaderPartitionId(uint32_t aPartitionId)
+{
+	mLeaderPartitionId = aPartitionId;
```

Comment:
would it be some kind of weird if partitionid set for device already joined the network would be delayed to take effect depending on some uncertain events? and what's more under this scenario, cli 'leaderpartitionid' would return an ineffective partitionid, other than the real partitionid the device belongs, does it sound a little confusing?